middle_end/flambda/compilenv_deps/patricia_tree.ml

```diff
@@ -420,7 +420,18 @@ end) = struct
     | Empty | Branch _ -> None
     | Leaf elt -> Some elt
 
-  let to_seq _ = Misc.fatal_error "to_seq not yet implemented"
+  let to_seq t =
+    let rec aux acc () =
+      match acc with
+      | [] -> Seq.Nil
+      | t0 :: r ->
+        begin match t0 with
+        | Empty -> aux r ()
+        | Leaf x -> Seq.Cons (x, aux r)
```

Comment:
Hmm, this should probably be tail recursive?

Reply:
I don't follow, here (i.e. line 430) we return a value, and that is somewhat tail-rec (since it cannot lead to more stacks). The recursive call, is actually the creation of a cloture (since `aux` takes 2 arguments), which is the delayed tail of the sequence.

Reply:
Oh yes, sorry, that is a partial application.  Fine.